content/blog/tidymodels-2023-q1/index.Rmd

```diff
@@ -0,0 +1,203 @@
+
+Since the beginning of 2021, we have been publishing [quarterly updates](https://www.tidyverse.org/categories/roundup/) here on the tidyverse blog summarizing what's new in the tidymodels ecosystem. The purpose of these regular posts is to share useful new features and any updates you may have missed. You can check out the [`tidymodels` tag](https://www.tidyverse.org/tags/tidymodels/) to find all tidymodels blog posts here, including our roundup posts as well as those that are more focused, like these posts from the past couple of months:
+
+-   [Tuning hyperparameters with tidymodels is a delight](https://www.tidyverse.org/blog/2023/04/tuning-delights/)
+-   [censored 0.2.0](https://www.tidyverse.org/blog/2023/04/censored-0-2-0/)
+-   [The tidymodels is getting a whole lot faster](https://www.simonpcouch.com/blog/speedups-2023/)
+
+Since [our last roundup post](https://www.tidyverse.org/blog/2022/12/tidymodels-2022-q4/), there have been CRAN releases of `r nrow(tm_data)` tidymodels packages. Here are links to their NEWS files:
+
+```{r}
+#| echo: FALSE
+#| results: asis
+
+cat(tm_data$news, sep = "\n")
+```
+
+We'll highlight a few especially notable changes below: more informative errors and faster code. First, loading the collection of packages:
+
+```{r}
+#| eval: FALSE
+library(tidymodels)
+library(embed)
+
+data("ames", package = "modeldata")
+```
+
+## More informative errors
+
+We like to make sure that the functions do what they are supposed to do. In the last months we are focused on making it so functions throw errors in such a way that they are easier for the users to pinpoint what went wrong and where. Since the modeling pipeline can be quite complicated, getting uninformative errors are a no-go.
+
+Across the tidymodels, error messages will now indicate the user-facing function that caused the error rather than the internal function that it came from.
+
+From dials, an error that looked like
+
+```r
+degree(range = c(1L, 5L))
+#> Error in `new_quant_param()`:
+#> ! Since `type = 'double'`, please use that data type for the range.
+```
+
+Now says that the error came from `degree()` rather than `new_quant_param()`
+
+```{r}
+#| error: TRUE
+degree(range = c(1L, 5L))
+```
+
+The same thing can be seen with the yardstick metrics
+
+``` r
+mtcars |>
+  accuracy(vs, am)
+#> Error in `dplyr::summarise()`:
+#> ℹ In argument: `.estimate = metric_fn(truth = vs, estimate = am, na_rm =
+#>   na_rm)`.
+#> Caused by error in `validate_class()`:
+#> ! `truth` should be a factor but a numeric was supplied.
+```
+
+which now errors much better
+
+```{r}
+#| error: TRUE
+mtcars |>
+  accuracy(vs, am)
+```
+
+Lastly, one of the biggest improvements came in recipes, which now shows which step caused the error instead of saying it happened in `prep()` or `bake()`. This is a huge improvement since you are likely to have many steps.
```

Comment:
```suggestion
Lastly, one of the biggest improvements came in recipes, which now shows which step caused the error instead of saying it happened in `prep()` or `bake()`. This is a huge improvement since preprocessing pipelines often which can string together many preprocessing steps.
```

Reply:
"often which can string" -> "which often string"

Reply:
Updated, thank you!